### PR TITLE
Dockerize master

### PR DESCRIPTION
**Pull request by robert-mcdermott**
@LisaMc this PR when merged with the master branch will allow us to create, test and deploy Oncoscape as a container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+MAINTAINER "Robert McDermott" robert.c.mcdermott@gmail.com
+
+# ADD the CRAN Repo to the apt sources list
+RUN echo "deb http://cran.fhcrc.org/bin/linux/ubuntu trusty/" > /etc/apt/sources.list.d/cran.fhcrc.org.list
+# Add the package verification key
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 51716619E084DAB9
+
+# Update the system and install packages
+RUN apt-get -y update && apt-get -y install \
```
Comment (LisaMc) [inline Dockerfile]: can you add these system dependencies within the Install.md file?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	m4 \
+	gcc \
+	g++ \
+	libxml2 \
+	libxml2-dev \
+        python-pip && \
+        pip install websocket-client 
+
+# create the sttrweb user and data directory
+RUN useradd -u 7534 -m -d /home/sttrweb -c "sttr web application" sttrweb && \
```
Comment (LisaMc) [inline Dockerfile]: does this mean it is specific to our system and directory structure? or is there a way to extract the full path and use a relative path or config file to generalize this?

File: Dockerfile (hunk 1 above)
Comment (robert-mcdermott) [inline Dockerfile]: @LisaMc yes I'll be updating the install documentation soon to cover platform specific instructions. For folks want to deploy Oncoscape via container, they don't need to worry about dependencies as they are all contained in the docker container.

File: Dockerfile (hunk 2 above)
Comment (robert-mcdermott) [inline Dockerfile]: @LisaMc if folks deploy Oncoscape via Docker then they don't need to worry about the directory structure inside the container; it has no relation to the system they are running the container on. As long as they have a Docker capable server, desktop or laptop (could be Windows, Linux or Mac) they can think of the Oncoscape container as a single large binary executable.

If folks want to provide their own datasets and libraries they can mount part of the host running Docker inside the container when the container is launched with a run time argument.

**Comment by robert-mcdermott** [comment]
@LisaMc is there something that I need to do before PR 53 and 54 can be merged?

Thanks,

-Robert

**Pull request merged**